nbcelltests/test.py

```diff
@@ -12,41 +12,81 @@
 import sys
 import subprocess
 import tempfile
+import ast
 from .define import TestMessage, TestType
-from .shared import extract_cellsources, extract_celltests, extract_extrametadata
+from .shared import extract_extrametadata
 from .tests_vendored import BASE, JSON_CONFD
 
+# TODO: eventually want assemble() and the rest to be doing something
+# better than building up code in strings. It's tricky to work with,
+# tricky to read, can't lint, can't test easily, unlikely to be
+# generally correct, etc etc :)
+
+# TODO: there are multiple definitions of "is tested", "is code cell"
+# throughout the code.
+
 INDENT = '    '
 
 
-def assemble_code(sources, tests):
-    cells = []
+def _is_empty(source):
+    # TODO: py2 utf8
+    return len(ast.parse(source).body) == 0
```

Comment:
Apart from the TODO already noted, this also would not handle magics - but it should.

Reply:
If it has a magic it won't be empty, so probably wrap in a try/except block?